lib/function.gi

```diff
@@ -78,13 +78,21 @@ end);
 
 InstallMethod(String, "for a function, with whitespace reduced", [IsFunction and IsInternalRep],
 function(fun)
-    local  s, stream;
+    local  s, stream,lines, pos, i;
     s := "";
     stream := OutputTextString(s, true);
```

Comment:
```suggestion
    stream := OutputTextString(s, true);
    SetPrintFormattingStatus(s, false);
```
And then add test which involve printing a function with a loooong string of whitespace, one which involves line wrapping because it is so long...

Reply:
There are other issues with this PR, but this is the most important one to fix, and I think fixing this will deal with many other issues.

Reply:
Interesting, I didn't expect that!

Still would be nice to have that test with a function that has a looooong whitespace string in it, just to make sure we don't regress on this in future refactorings.

Reply:
I do have a test (I make the string with concatenation, just because long tests can look akrward)

Reply:
Ok, but that test is not yet in here, is it? At least I don't see it